github-actions/pulumi-python.yml

```diff
@@ -41,30 +41,39 @@ jobs:
       - uses: actions/checkout@v2
 
       - name: Install Python
-        uses: actions/setup-python@v1
+        uses: actions/setup-python@v2
         with:
           python-version: 3.6.x
-    
-      - run: |
-          # Check if should use venv
-          PULUMI_VENV=$(cat Pulumi.yaml | grep "virtualenv:" | cut -d':' -f2)
-          if [ -z ${PULUMI_VENV} ]; then
-              pip3 install -r requirements.txt
-          else
-              python3 -m venv ${PULUMI_VENV}
-              ${PULUMI_VENV}/bin/python -m pip install --upgrade pip setuptools wheel
-              ${PULUMI_VENV}/bin/python -m pip install -r requirements.txt
-          fi
-        name: Restore Python dependencies
-        working-directory: ${{ env.PULUMI_WORKING_DIRECTORY }}
 
-      - name: Install Pulumi CLI
-        uses: pulumi/action-install-pulumi-cli@v1.0.1
+      - run: pip install -r requirements.txt
```

Comment:
I realize this works too but shouldn't we ensure a `venv` is activated first? That's what the previous step was doing.

Reply:
It's a good question - personally, I only use venv locally to stop my development machine having its state changed. In an ephemeral CI environment, I never care as the machine is built for a single purpose

I think the previous statement made things look more complex but I can certainly add it back in if you (and @infin8x) feel its important

Reply:
> In an ephemeral CI environment, I never care as the machine is built for a single purpose

Yep. That's what I was thinking too, but then thought about self-hosted runners. But maybe that's not the most common case. I am fine with this if you feel this is the common workflow for Python users in CI.

Reply:
Given GitHub Actions' heavy bias towards hosted runners, I'm OK with this being what's in the template.